src/Text/Pandoc/Readers/Docx.hs

```diff
@@ -494,15 +497,30 @@ singleParaToPlain blks
       singleton $ Plain ils
 singleParaToPlain blks = blks
 
-cellToBlocks :: PandocMonad m => Docx.Cell -> DocxContext m Blocks
-cellToBlocks (Docx.Cell bps) = do
+cellToCell :: PandocMonad m => RowSpan -> Docx.Cell -> DocxContext m Pandoc.Cell
+cellToCell rowSpan (Docx.Cell gridSpan _ bps) = do
   blks <- smushBlocks <$> mapM bodyPartToBlocks bps
-  return $ fromList $ blocksToDefinitions $ blocksToBullets $ toList blks
-
-rowToBlocksList :: PandocMonad m => Docx.Row -> DocxContext m [Blocks]
-rowToBlocksList (Docx.Row cells) = do
-  blksList <- mapM cellToBlocks cells
-  return $ map singleParaToPlain blksList
+  let blks' = singleParaToPlain $ fromList $ blocksToDefinitions $ blocksToBullets $ toList blks
+  return (cell AlignDefault rowSpan (ColSpan (fromIntegral gridSpan)) blks')
+
+rowToRow :: PandocMonad m => Docx.Row -> DocxContext m Pandoc.Row
+rowToRow (Docx.Row _ cells) = do
+  cells' <- traverse (cellToCell (RowSpan 1)) cells
+  return (Pandoc.Row nullAttr (cells'))
+
+rowsToRows :: PandocMonad m => [Docx.Row] -> DocxContext m [Pandoc.Row]
+rowsToRows rows = do
+  let rowspans = (fmap . fmap) RowSpan (Docx.rowsToRowspans rows)
+  let rowCells = fmap (\(Docx.Row _ cs) -> cs) rows
+  cells <- zipWithM (zipWithM cellToCell) rowspans rowCells
```

Comment:
I added a `trace` function here, and I think `rowspans` and `rowCells` have different lengths, so that's why cells at the end are coming up empty.

Reply:
Yup, absolutely! Sorry, I didn't see this, just came back here to point that out. The problem is my `rowsToRowspans` function, which uses `transpose` to look at the columns and then `transposes` again to get back to rows: unfortunately that only works when every row has the same number of elements, which is not true here. I'll write a `transpose` that takes the column spans into account I reckon, that should sort that.